frame/balances/src/lib.rs

```diff
@@ -784,7 +784,7 @@ pub mod pallet {
 				return false
 			}
 			a.flags.set_new_logic();
-			if !a.reserved.is_zero() || !a.frozen.is_zero() {
+			if !a.reserved.is_zero() && a.frozen.is_zero() {
```

Comment:
This logic here is essentially saying that frozen funds don't contribute any provider refs; is this true?

Reply:
Well, it's that frozen funds already require a consumer ref in the original logic, so if we have non-zero frozen funds, then we already have the consumer we need (and if we have a consumer, we must have a provider).

Reply:
Ah ok, so it's reserved funds in the old logic that did not have a consumer, and so we have to add it here?